### PR TITLE
Migrate `Nested` test

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
Hmm interesting. Only fails on native. I was only running the JVM variant locally when migrating.

**Comment by JakeWharton** [comment]
It looks like the override isn't working on native. When I force step into `equals` it goes to `kotlin_Any_equals` symbol.

**Comment by JakeWharton** [comment]
I dumped the LLVM bitcode and then converted it back to LLVM assembly to confirm that the `poko.Nested.equals()` function is completely absent from the native compilation. Its `hashCode` and `toString` are present, but something weird is happening with `equals`. From logging in the compiler plugin I can see that it does match on the fake `equals` and attempts to replace its body, but for some reason that doesn't end up in the binary and I don't know why.

File: poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/PokoMembersTransformer.kt [hunk 1]
```diff
@@ -130,7 +131,10 @@ internal class PokoMembersTransformer(
             generateFunctionBody(properties)
         }
 
-        reflectivelySetFakeOverride(false)
+        (this as IrSimpleFunction).apply {
+            isFakeOverride = false
+            isExternal = false
```
Comment (JakeWharton) [inline poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/PokoMembersTransformer.kt:136]: > In native Any.equals has external flag. Maybe you accidentially copied it somewhere?

Saved by JetBrains, again.

https://kotlinlang.slack.com/archives/C7L3JB43G/p1689958980460969?thread_ts=1689958221.929309&cid=C7L3JB43G

Comment (drewhamilton) [reply poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/PokoMembersTransformer.kt:136]: Amazing. What does "external" mean here?

Comment (drewhamilton) [reply poko-compiler-plugin/src/main/kotlin/dev/drewhamilton/poko/ir/PokoMembersTransformer.kt:136]: Answered in the Slack thread; it means the function is implemented somewhere other than source code.

**Review by drewhamilton — APPROVED** [review]

**Pull request merged**